llvm/lib/IR/EHPersonalities.cpp

```diff
@@ -25,7 +26,14 @@ EHPersonality llvm::classifyEHPersonality(const Value *Pers) {
       Pers ? dyn_cast<GlobalValue>(Pers->stripPointerCasts()) : nullptr;
   if (!F || !F->getValueType() || !F->getValueType()->isFunctionTy())
     return EHPersonality::Unknown;
-  return StringSwitch<EHPersonality>(F->getName())
+
+  std::string Name = F->getName().str();
+  if (Triple(F->getParent()->getTargetTriple()).isWindowsArm64EC()) {
+    if (auto DemangledName = getArm64ECDemangledFunctionName(Name))
+      Name = *DemangledName;
```

Comment:
Since we're recognizing only C symbols here anyway, I guess we could use `StringRef` for the name and `consume_front("#")` here.